src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -143,9 +144,9 @@ private static ExternalProjectReference GetExternalProject(PackageSpec rootProje
             {
                 settings = restoreArgs.GetSettings(rootPath);
             }
-
-            var globalPath = restoreArgs.GetEffectiveGlobalPackagesFolder(rootPath, settings);
-            var fallbackPaths = restoreArgs.GetEffectiveFallbackPackageFolders(settings);
+            // All of these are set when the DG Spec is created!
+            var fallbackPaths = new ReadOnlyCollection<string>(project.PackageSpec.RestoreMetadata.FallbackFolders);
+            var globalPath = project.PackageSpec.RestoreMetadata.PackagesPath;
```

Comment:
If this is empty it should be set to the default using SettingsUtility with NullSettings.Instance as the settings.

The global path is only input that must exist. You could have zero sources and everything in the global folder, in which case restore works. But if you have no global folder you can't do anything.